azure-pipelines.yml

```diff
@@ -3,19 +3,64 @@
 
 # Build STL targeting x86, x64, arm, arm64
 
-jobs:
-- template: azure-devops/run-build.yml
-  parameters:
-    targetPlatform: x86
+variables:
+  agentPool: 'StlBuild-2020-04-04-3'
 
-- template: azure-devops/run-build.yml
-  parameters:
-    targetPlatform: x64
+stages:
+  - stage: Code_Format
+    displayName: 'Code Format'
+    jobs:
+      - job: Code_Format_Validation
+        timeoutInMinutes: 90
+        displayName: 'Validation'
+        pool:
+          name: $(agentPool)
 
-- template: azure-devops/run-build.yml
-  parameters:
-    targetPlatform: arm
+        steps:
+          - checkout: self
+            clean: true
+            submodules: true
+          - script: |
+              call "%PROGRAMFILES(X86)%\Microsoft Visual Studio\2019\Preview\Common7\Tools\VsDevCmd.bat" ^
+              -arch=amd64 -host_arch=amd64 -no_logo
+              cmake -G Ninja -DCMAKE_CXX_COMPILER=cl -DCMAKE_BUILD_TYPE=Release ^
+              -S $(Build.SourcesDirectory)\tools -B $(Build.ArtifactStagingDirectory)\tools
+              cd $(Build.ArtifactStagingDirectory)\tools
+              cmake --build .
+            displayName: 'Build Support Tools'

```

Comment:
Should we take steps to get these on the box in advance now that we have creation of VMs scripted? (I'm asking if we should file an issue, not saying you need to make that change here)

Reply:
meh, idk, I think it isn't worth the hassle considering how fast the build is.

Reply:
It's not a 'how fast the build is' consideration it's a 'we have a lot of supporting infrastructure to build these first' consideration. But I agree it isn't worth opening that can of worms unless you want.

Reply:
What supporting infrastructure are you referring to? The developer overhead of having more YAML?

Reply:
I think it's highly desirable to build `validate.exe` as part of the checks, so it can be updated by PRs without rebuilding the VMs. Similarly for `parallelize.exe` which we also had to teach to skip files. Therefore, I believe that this is good as-is.